pkg/source/chirpstack/config/config.go

```diff
@@ -32,96 +34,113 @@ import (
+
+	apiKey, caCertPath, url, joinEUI string
+	flags                            *pflag.FlagSet
+	FPStore                          *frequencyplans.Store
+	insecure                         bool
+
+	ClientConn *grpc.ClientConn
 
-	flags.StringVar(&config.url,
+	ExportVars,
+	ExportSession bool
+	FrequencyPlanID string
+	JoinEUI         *types.EUI64
+}
+
+func New() *Config {
+	config := &Config{
+		flags: &pflag.FlagSet{},
+	}
+
+	config.flags.StringVar(&config.url,
 		"api-url",
-		os.Getenv("CHIRPSTACK_API_URL"),
+		"",
 		"ChirpStack API URL")
-	flags.StringVar(&config.token,
-		"api-token",
-		os.Getenv("CHIRPSTACK_API_TOKEN"),
-		"ChirpStack API Token")
-	flags.StringVar(&config.caPath,
-		"api-ca",
-		os.Getenv("CHIRPSTACK_API_CA"),
-		"(optional) CA for TLS")
-	flags.BoolVar(&config.insecure,
-		"api-insecure",
-		os.Getenv("CHIRPSTACK_API_INSECURE") == "1",
+	config.flags.StringVar(&config.apiKey,
+		"api-key",
+		"",
+		"ChirpStack API key")
+	config.flags.StringVar(&config.caCertPath,
+		"ca-cert-path",
+		"",
+		"(optional) Path to the CA certificate file for ChirpStack API TLS connections")
+	config.flags.BoolVar(&config.insecure,
+		"insecure",
+		false,
 		"Do not connect to ChirpStack over TLS")
-	flags.BoolVar(&config.ExportVars,
+	config.flags.BoolVar(&config.ExportVars,
 		"export-vars",
 		false,
 		"Export device variables from ChirpStack")
-	flags.BoolVar(&config.ExportSession,
+	config.flags.BoolVar(&config.ExportSession,
 		"export-session",
-		true,
+		false,
 		"Export device session keys from ChirpStack")
-	flags.StringVar(&config.joinEUI,
+	config.flags.StringVar(&config.joinEUI,
 		"join-eui",
-		os.Getenv("JOIN_EUI"),
+		"",
 		"JoinEUI of exported devices")
-	flags.StringVar(&config.FrequencyPlanID,
+	config.flags.StringVar(&config.FrequencyPlanID,
 		"frequency-plan-id",
-		os.Getenv("FREQUENCY_PLAN_ID"),
+		"",
 		"Frequency Plan ID of exported devices")
 
-	return config, flags
+	return config
 }
 
-type Config struct {
-	source.Config
-
-	ClientConn *grpc.ClientConn
+func (c *Config) Initialize(src source.Config) error {
+	c.src = src
 
-	token, caPath, url,
-	FrequencyPlanID string
-
-	joinEUI string
-	JoinEUI *types.EUI64
-
-	insecure,
-	ExportVars,
-	ExportSession bool
-}
-
-func (c *Config) Initialize() error {
-	if c.token == "" {
+	if c.apiKey = os.Getenv("CHIRPSTACK_API_KEY"); c.apiKey == "" {
 		return errNoAPIToken.New()
 	}
-	if c.url == "" {
+	if c.url = os.Getenv("CHIRPSTACK_API_URL"); c.url == "" {
 		return errNoAPIURL.New()
 	}
-	if c.FrequencyPlanID == "" {
+	if c.FrequencyPlanID = os.Getenv("FREQUENCY_PLAN_ID"); c.FrequencyPlanID == "" {
 		return errNoFrequencyPlan.New()
 	}
-
+	if c.joinEUI = os.Getenv("JOIN_EUI"); c.joinEUI == "" {
+		return errNoJoinEUI.New()
+	}
 	c.JoinEUI = &types.EUI64{}
 	if err := c.JoinEUI.UnmarshalText([]byte(c.joinEUI)); err != nil {
 		return errInvalidJoinEUI.WithAttributes("join_eui", c.joinEUI)
 	}
+	c.caCertPath = os.Getenv("CHIRPSTACK_CA_CERT_PATH")
```

Comment:
Won't this always override the CLI flags ?

Reply:
Yes indeed. We do this for all the other sources the same currently. I think we should fix this separately for all the sources.

Reply:
I would like to see this fixed everywhere before we make a new release, as it is very broken. 

The change itself comes from https://github.com/TheThingsNetwork/lorawan-stack-migrate/pull/97#discussion_r1378945712 , but the idea is that only secrets shouldn't be initialized as flag defaults - the rest of the setup does not need this.

Reply:
Sounds good to me. I'll file a follow up before we make a release.